subprojects/base-services/src/main/java/org/gradle/internal/classloader/ClassLoaderUtils.java

```diff
@@ -101,6 +89,63 @@ public static void disableUrlConnectionCaching() {
     }
 
     public static <T> Class<T> define(ClassLoader targetClassLoader, String className, byte[] clazzBytes) {
-        return Cast.uncheckedCast(UNSAFE.defineClass(className, clazzBytes, 0, clazzBytes.length, targetClassLoader, null));
+        return CLASS_DEFINER.injectClass(targetClassLoader, className, clazzBytes);
+    }
+
+    private interface ClassDefiner {
+        <T> Class<T> injectClass(ClassLoader classLoader, String className, byte[] classBytes);
```

Comment:
The method should also be renamed to `defineClass`, shouldn't it?

Reply:
It should, thanks.